include/robocup_client/communication/socket/base_socket.hpp

```diff
@@ -18,33 +18,34 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-#ifndef ROBOCUP_CLIENT__ROBOT_CLIENT_HPP_
-#define ROBOCUP_CLIENT__ROBOT_CLIENT_HPP_
-
-#include <musen/musen.hpp>
-
-#include <string>
-#include <memory>
-
-#include "./messages.pb.h"
+#ifndef ROBOCUP_CLIENT__COMMUNICATION__SOCKET__BASE_SOCKET_HPP_
+#define ROBOCUP_CLIENT__COMMUNICATION__SOCKET__BASE_SOCKET_HPP_
 
 namespace robocup_client
 {
 
-class RobotClient : public musen::Client
+namespace communication
+{
+
+class BaseSocket
 {
 public:
-  explicit RobotClient(
-    const std::string & host, const int & port,
-    std::shared_ptr<musen::TcpSocket> tcp_socket = std::make_shared<musen::TcpSocket>());
+  BaseSocket();
+  ~BaseSocket();
+
+  virtual bool connect();
+  virtual bool disconnect();
 
-  bool connect();
+  const int & get_sockfd() const;
```

Comment:
Please don't pass constant reference (`cont T &`) for primitive type (`bool`, `int`, `float`, etc.).

Reason: basically constant reference is good for passing value between function without the need to reallocating spaces. But in the case of primitive type, which size is usually less than pointer size (32/64 bit), it will become less efficient.

Example: passing `char` dirrectly will cause program to allocate 8bit, while passing it as constant reference (or pointer) will cause it to allocate 32/64bit, not to mention the need to get the real value from address that's pointed by the pointer.

> See this problem in other lines too!

Reply:
done, mas